keanu-python/keanu/algorithm/sampling.py

```diff
@@ -17,31 +16,90 @@
 java_import(k.jvm_view(), "io.improbable.keanu.algorithms.mcmc.nuts.NUTS")
 java_import(k.jvm_view(), "io.improbable.keanu.algorithms.mcmc.Hamiltonian")
 
-algorithms = {
-    'metropolis': k.jvm_view().MetropolisHastings,
-    'NUTS': k.jvm_view().NUTS,
-    'hamiltonian': k.jvm_view().Hamiltonian
-}
+
+class PosteriorSamplingAlgorithm:
+
+    def __init__(self, sampler: JavaObject):
+        self._sampler = sampler
+
+    def get_sampler(self) -> JavaObject:
+        return self._sampler
+
+
+class MetropolisHastingsSampler(PosteriorSamplingAlgorithm):
+
+    def __init__(self,
+                 proposal_distribution: str = None,
```

Comment:
I know that the `.builder()` sets all of the defaults defined in Java but they feel very hidden from a python user. What're your thoughts on adding all the defaults to the arguments here too? Means we have another thing to monitor across both API's but it'll be clearer for python users

Reply:
As discussed - we should raise a ticket to document these things in Python better.  Deferring that till then.